src/app/clusters/wifi-network-diagnostics-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.h>
+#include <app/static-cluster-config/WiFiNetworkDiagnostics.h>
+#include <app/util/attribute-storage.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::WiFiNetworkDiagnostics;
+using namespace chip::app::Clusters::WiFiNetworkDiagnostics::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+// for fixed endpoint, this file is ever only included IF WiFi network diagnostics is enabled and that MUST happen only on endpoint
+// 0 the static assert is skipped in case of dynamic endpoints.
+static_assert((WiFiNetworkDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 1 &&
+               WiFiNetworkDiagnostics::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId) ||
```

Comment:
This is NOT correct: diagnostics clusters are not singletons. A device may have multiple interfaces and each may have their own diagnostics.

Now our implementation is 100% buggy and we support a single delegate so we do not support that properly. However the spec does say we support it...

Reply:
This is covered for example in the device library `2.8. Secondary Network Interface`

Reply:
Removed that assert and filed #40317